src/ts/ssh-tcp/services/portForwardingService.ts

```diff
@@ -61,8 +61,25 @@ export class PortForwardingService extends SshService {
 	public static readonly portForwardChannelType = 'forwarded-tcpip';
 	public static readonly reversePortForwardChannelType = 'direct-tcpip';
 
+	/**
+	 * Maps from local IP address and FORWARDED port number to the object that manages
+	 * listening for incoming connections on that port and forwarding them through the session.
+	 *
+	 * Note the actual local source port number used may be different from the forwarded port
+	 * number if the local TCP listener factory chose a different port. The forwarded port number
+	 * is used to identify the port in any messages exchanged between client and server.
+	 */
 	private readonly localForwarders = new Map<string, LocalPortForwarder>();
-	private readonly remoteConnectors = new Map<string, RemotePortConnector>();
+
+	/**
+	 * Maps from FORWARDED port numbers to the object that manages relaying forwarded connections
```

Comment:
`localForwarders` map uses both IP address and forwarded port as the key, and `remoteConnectors` only uses forwarded port. Can it be that the same forwarded port number is forwarded from different local IPs by different local forwarders but would only have one `RemotePortConnector` because `remoteConnectors` key is just the port number?

Reply:
> Can it be that the same forwarded port number is forwarded from different local IPs by different local forwarders

It's not possible because it would be ambiguous. After the port is forwarded, when the other side gets a connection it will send a PortForwardChannelOpenMessage with just the port number as the destination. There cannot be multiple destinations for channels associated with the same port number. Anyway the other side doesn't have any information or control over the destination; it is up to the side that requested forwarding to route the incoming channel connections.

And now that I think about it, the `localForwarders` map can use only the port number as key also. I'll update that. It was needlessly complicated before due to an incomplete understanding of the protocol spec when I originally wrote the code.

Reply:
Updated `localForwarders` to use a simple port key so the 2 maps are consistent.